lib/dnsUtils.js

```diff
@@ -68,10 +69,13 @@ class DNSutils {
       dns.flags |= packet.AUTHENTIC_DATA
       dns.additionals[0].flags |= packet.DNSSEC_OK
     }
-    if (opts.ecs != null) {
-      dns.additionals.options = [{
+    if (opts.ecs != null || net.isIP(opts.subnet) !== 0) {
+      // https://tools.ietf.org/html/rfc7871#section-11.1
+      const prefix = opts.subnet && net.isIPv4(opts.subnet) ? 24 : 56
+      dns.additionals[0].options = [{
         code: 'CLIENT_SUBNET',
-        sourcePrefixLength: opts.ecs
+        ip: opts.subnet || '0.0.0.0',
```

Comment:
or should I use `dns.getServers()` instead?

Reply:
I think the right thing is the IP address of your machine *after* NAT; your external IP address.  I don't think dns.getServers() is much better, though, because those are often addresses local to your network as well.

Let's default to quad-0 so we don't leak private info without the user's involvement, then add an enhancement request to think about getting the external address by pinging some API on a well-known IP.